modin/polars/dataframe.py

```diff
@@ -0,0 +1,1340 @@
+
+        Args:
+            other: DataFrame to join with.
+            left_on: Column to join on in the left DataFrame.
+            right_on: Column to join on in the right DataFrame.
+            on: Column to join on in both DataFrames.
+            by_left: Columns to join on in the left DataFrame.
+            by_right: Columns to join on in the right DataFrame.
+            by: Columns to join on in both DataFrames.
+            strategy: Strategy to use for the join.
+            suffix: Suffix to add to the columns.
+            tolerance: Tolerance for the join.
+
+        Returns:
+            Joined DataFrame.
+        """
+        if on is not None and left_on is None and right_on is None:
+            left_on = right_on = on
+        if by is not None and by_left is None and by_right is None:
+            by_left = by_right = by
+        return self.__constructor__(
+            _query_compiler=self._query_compiler.merge_asof(
+                other._query_compiler,
+                left_on=left_on,
+                right_on=right_on,
+                left_by=by_left,
+                right_by=by_right,
+                direction=strategy,
+                suffixes=("", suffix),
+                tolerance=tolerance,
+            )
+        )
+
+    def melt(
+        self,
+        id_vars=None,
+        value_vars=None,
+        variable_name: str | None = None,
+        value_name: str | None = None,
+    ) -> "DataFrame":
+        """
+        Melt the DataFrame.
+
+        Args:
+            id_vars: Columns to keep.
+            value_vars: Columns to melt.
+            variable_name: Name of the variable column.
+            value_name: Name of the value column.
+
+        Returns:
+            Melted DataFrame.
+        """
+        return self.__constructor__(
+            _query_compiler=self._query_compiler.melt(
+                id_vars=id_vars,
+                value_vars=value_vars,
+                var_name=variable_name,
+                value_name=value_name,
+            )
+        )
+
+    def merge_sorted(self, other: "DataFrame", on: str | list[str]) -> "DataFrame":
+        # TODO: support natural join + sort
+        raise NotImplementedError("not yet")
+
+    def partition_by(
+        self,
+        by,
+        *more_by,
+        maintain_order: bool = True,
+        include_key: bool = True,
+        as_dict: bool = False,
+    ) -> list["DataFrame"] | dict[Any, "DataFrame"]:
+        """
+        Partition the DataFrame by the given columns.
+
+        Args:
+            by: Columns to partition by.
+            more_by: Additional columns to partition by.
+            maintain_order: Whether to maintain the order of the partitions.
+            include_key: Whether to include the partition key.
+            as_dict: Whether to return the partitions as a dictionary.
+
+        Returns:
+            List of DataFrames or dictionary of DataFrames.
+        """
+        if isinstance(by, str):
```

Comment:
Do we need to deal with `maintain_order`?

Reply:
Generally Modin maintains the order of the input, so I didn't opt to try to match polars order when `maintain_order=False`. That should be left for the future.